src/NLog.Web.AspNetCore/NLogRequestPostedBodyMiddleware.cs

```diff
@@ -0,0 +1,138 @@
+            _configuration = configuration;
+        }
+
+        /// <summary>
+        /// This allows interception of the HTTP pipeline for logging purposes
+        /// </summary>
+        /// <param name="context">The HttpContext</param>
+        /// <param name="next">The RequestDelegate that is to be executed next in the HTTP pipeline</param>
+        /// <returns></returns>
+        public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+        {
+            // Perform null checking
+            if (context.Request == null)
+            {
+                InternalLogger.Debug("NLogRequestPostedBodyMiddleware: HttpContext.Request stream is null");
+                // Execute the next class in the HTTP pipeline, this can be the next middleware or the actual handler
+                await next(context).ConfigureAwait(false);
+                return;
+            }
+
+            // Perform null checking
+            if (context.Request.Body == null)
+            {
+                InternalLogger.Debug("NLogRequestPostedBodyMiddleware: HttpContext.Request.Body stream is null");
+                // Execute the next class in the HTTP pipeline, this can be the next middleware or the actual handler
+                await next(context).ConfigureAwait(false);
+                return;
+            }
+
+            // If we cannot read the stream we cannot capture the body
+            if (!context.Request.Body.CanRead)
+            {
+                InternalLogger.Debug("NLogRequestPostedBodyMiddleware: HttpContext.Request.Body stream is non-readable");
+                // Execute the next class in the HTTP pipeline, this can be the next middleware or the actual handler
+                await next(context).ConfigureAwait(false);
+                return;
+            }
+
+            // This is required, otherwise reading the request will destructively read the request
+            context.Request.EnableBuffering();
+
+            // If we cannot reset the stream position to zero, and then back to the original position
+            // we cannot capture the body
+            if (!context.Request.Body.CanSeek)
+            {
+                InternalLogger.Debug("NLogRequestPostedBodyMiddleware: HttpContext.Request.Body stream is non-seekable");
+                // Execute the next class in the HTTP pipeline, this can be the next middleware or the actual handler
+                await next(context).ConfigureAwait(false);
+                return;
+            }
+
+            // Use the predicate in the configuration instance that takes the HttpContext as an argument
+            if (_configuration.ShouldCapture(context))
+            {
+                // Save the POST request body in HttpContext.Items with a key of '__nlog-aspnet-request-posted-body'
+                context.Items[AspNetRequestPostedBodyLayoutRenderer.NLogPostedRequestBodyKey] =
+                    await GetString(context?.Request.Body).ConfigureAwait(false);
```

Comment:
When `GetString(bodyStream)` returns null or string.Empty, then it should not insert into items

Reply:
completed as requested